src/main/java/org/springframework/integration/kafka/listener/QueueingMessageListenerInvoker.java

```diff
@@ -78,14 +95,8 @@ public void enqueue(KafkaMessage message) {
 			// handle the case when the thread is interrupted while the adapter is still running
 			// retry adding the message to the queue until either we succeed, or the adapter is stopped
 			while (!added && this.running) {
-				try {
-					this.messages.put(message);
-					added = true;
-				}
-				catch (InterruptedException e) {
-					// we ignore the interruption signal if we are still running, but pass it on if we are stopped
-					wasInterruptedWhileRunning = true;
-				}
+				ringBufferProcessor.onNext(message);
+				added = true;
```

Comment:
So, what you are saying must looks like:
````java
public void enqueue(KafkaMessage message) {
	boolean wasInterruptedWhileRunning = false;
	if (this.running) {
		boolean added = false;
		// handle the case when the thread is interrupted while the adapter is still running
		// retry adding the message to the queue until either we succeed, or the adapter is stopped
		while (!added && this.running) {
			try {
				ringBufferProcessor.onNext(message);
				added = true;
			}
			catch (InterruptedException e) {
				// we ignore the interruption signal if we are still running, but pass it on if we are stopped
				wasInterruptedWhileRunning = true;
			}
		}
	}
	if (wasInterruptedWhileRunning) {
		Thread.currentThread().interrupt();
	}
}
````
Don't you want to say that this is a typical pattern to get deal with `RingBuffer` ?
The blocking code looks like:
````java
while (wrapPoint > (minSequence = Util.getMinimumSequence(gatingSequences, nextValue))) {
      LockSupport.parkNanos(1L); // TODO: Use waitStrategy to spin
}
````
So, it isn't *blocking* really...

Have I missed anything?